src/internet_identity/src/assets.rs

```diff
@@ -86,61 +78,104 @@ pub fn init_assets() {
-            ContentEncoding::Identity,
-            ContentType::OCTETSTREAM,
-        ),
-    ]
+static ASSET_DIR: Dir<'_> = include_dir!("$CARGO_MANIFEST_DIR/../../dist");
+
+// Gets the static assets. All static assets are prepared only once (like injecting the canister ID).
+fn get_static_assets() -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets = collect_assets_recursive(&ASSET_DIR);
+
+    // Required to make II available on the identity.internetcomputer.org domain.
+    // See https://internetcomputer.org/docs/current/developer-docs/production/custom-domain/#custom-domains-on-the-boundary-nodes
+    assets.push((
+        "/.well-known/ic-domains".to_string(),
+        b"identity.internetcomputer.org".to_vec(),
+        ContentEncoding::Identity,
+        ContentType::OCTETSTREAM,
+    ));
+
+    assets
+}
+
+fn collect_assets_recursive(dir: &Dir) -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets = collect_assets_from_dir(dir);
+    for subdir in dir.dirs() {
+        assets.extend(collect_assets_recursive(subdir).into_iter());
+    }
+    assets
+}
+
+fn collect_assets_from_dir(dir: &Dir) -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets: Vec<(String, Vec<u8>, ContentEncoding, ContentType)> = vec![];
+    for asset in dir.files() {
+        let file_bytes = asset.contents().to_vec();
+        let (content, encoding, content_type) = match file_extension(asset) {
+            "css" => (file_bytes, ContentEncoding::Identity, ContentType::CSS),
+            "html" => (
+                fixup_html(String::from_utf8_lossy(&file_bytes).as_ref())
+                    .as_bytes()
+                    .to_vec(),
+                ContentEncoding::Identity,
+                ContentType::HTML,
+            ),
+            "ico" => (file_bytes, ContentEncoding::Identity, ContentType::ICO),
+            "js.gz" => (file_bytes, ContentEncoding::GZip, ContentType::JS),
+            "png" => (file_bytes, ContentEncoding::Identity, ContentType::PNG),
+            "webp" => (file_bytes, ContentEncoding::Identity, ContentType::WEBP),
+            _ => panic!("Unknown asset type: {}", asset.path().display()),
+        };
+
+        assets.push((file_to_asset_path(asset), content, encoding, content_type));
+    }
+    assets
+}
+
+/// Returns the portion of the filename after the first dot.
+/// This corresponds to the file extension for the assets handled by this canister.
+///
+/// The builtin `extension` method on `Path` does not work for file extensions with multiple dots
+/// such as `.js.gz`.
+fn file_extension<'a>(asset: &'a File) -> &'a str {
+    asset
+        .path()
+        .file_name()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .split_once('.')
+        .unwrap()
+        .1
+}
+
+/// Returns the asset path for a given file:
+/// * make relative path absolute
+/// * map **/index.html to **/
+/// * map **/<foo>.html to **/foo
```

Comment:
@nmattia should we support trailing slash strategy as well or just don't forget that we don't?

browser url `/foo` -> asset `/foo/index.html`

Reply:
Up to you and @frederikrothenberger really!

Reply:
@peterpeterparker: Could you clarify: Does your comment mean that rather than serving `about.html` on `/about`, we should instead serve `about/index.html`?

I.e. we move the emitted `about.html`?

Reply:
No no nothing to be changed. With my comment I just meant that it works for now but if we ever change the build process to a trailing slash strategy (`./dist/foo/index.html` instead of `./dist/foo.html/), it might no work. Therefore asking if we already want to support this or just remember that if we ever do so, we have to extend the mapping.

Reply:
Ah, ok. In that case, I would leave it as is. Once we change, it will break in obvious ways, so even if we forget, we should not loose too much time. And then we can change the mapping.

Reply:
Works for me